src/shared/UniversalRender.jsx

```diff
@@ -64,15 +69,23 @@ const scrollTop = (el, topOffset, prevDocumentInfo, triesRemaining) => {
     const documentInfo = {
         scrollHeight: document.body.scrollHeight,
         scrollTop: Math.ceil(document.scrollingElement.scrollTop),
-        scrollTarget: calcOffsetRoot(el) + topOffset
+        scrollTarget: calcOffsetRoot(el) + topOffset,
+        direction: prevDocumentInfo.direction,
     };
-
-    if(prevDocumentInfo.scrollTop > (documentInfo.scrollTop + SCROLL_UP_FUDGE_PIXELS)) { //detecting that the user has scrolled in an up direction
-        return;
+    let doScroll = false;
+    if(SCROLL_DIRECTION_DOWN === prevDocumentInfo.direction) {
+        doScroll = (!(prevDocumentInfo.scrollTop > (documentInfo.scrollTop + SCROLL_FUDGE_PIXELS))
+            && (documentInfo.scrollTop < documentInfo.scrollTarget
+                || prevDocumentInfo.scrollTarget < documentInfo.scrollTarget
+                || prevDocumentInfo.scrollHeight < documentInfo.scrollHeight));
+    } else if(SCROLL_DIRECTION_UP === prevDocumentInfo.direction) {
```

Comment:
again, please don't try to do the interpreter's work for it. this variable is `const`, it cannot be redefined, and the Yoda syntax is really confusing

Reply:
See above oh misunderstanding one.

Reply:
looks like prevDocumentInfo is a function parameter

we should be getting linter errors if we mutate function parameters, but we've disabled that rule :( https://github.com/airbnb/javascript#functions--mutate-params

Reply:
@bnchdrff omg; we need to enable that. I still prefer Yoda style, but that rule would salve my offended sensibilities just a bit.